espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/MobileBackdoor.java

```diff
@@ -0,0 +1,75 @@
+package io.appium.espressoserver.lib.handlers;
+
+import android.app.Activity;
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidArgumentException;
+import io.appium.espressoserver.lib.helpers.ActivityHelper;
+import io.appium.espressoserver.lib.helpers.InvocationOperation;
+import io.appium.espressoserver.lib.model.MobileBackdoorMethod;
+import io.appium.espressoserver.lib.model.MobileBackdoorParams;
+
+import static io.appium.espressoserver.lib.helpers.AndroidLogger.logger;
+
+public class MobileBackdoor implements RequestHandler<MobileBackdoorParams, String> {
+
+    @Override
+    public String handle(final MobileBackdoorParams params) throws AppiumException {
+        logger.info("Invoking Backdoor");
+        if (params.getTarget() == null) {
+            throw new InvalidArgumentException("Target must not be empty and must be of type: 'activity', 'application'");
+        }
+
+        Activity activity = ActivityHelper.getCurrentActivity();
+        List<InvocationOperation> ops = getBackdoorOperations(params);
+        Object invocationResult;
+        switch (params.getTarget()) {
+            case ACTIVITY:
+                invocationResult = invokeBackdoorMethods(activity, ops);
+                break;
+            case APPLICATION:
+                invocationResult = invokeBackdoorMethods(activity.getApplication(), ops);
+                break;
+            default:
+                throw new InvalidArgumentException(String.format("target cannot be %s", params.getTarget()));
+        }
+
+        return invocationResult == null ? null : invocationResult.toString();
```

Comment:
Is it necessary to transform the result to string?

Reply:
I'd let GSON handle the serialization

Reply:
Not necessary, Makes complete sense to return actual Object.